03. An EDSL for Hard-working IT Programmers.md

```diff
@@ -1,145 +1,172 @@
+Object-oriented Programming naturally fit with this concept of inversion of control, which pervades IT problems. Whenever there is more than one asynchronous input, there is multitasking or inversion of control with shared state. The solution is a state machine. What OOP does is to split this state machine into smaller state machines called "objects" that interact with each other. But that implies the need to "deconstruct" the specification.
 
-In OOP, you can not create a function or method called "fry_the_eggs" that stop the fire when the eggs are fried, without blocking the execution of everithing else. So you need to manage explicitly at least two threads that sooner or later will have to communicate asynchronouslly. Alternatively, you may define start-fire, stop-fire and a callback called egg-fried so that you wire-up the application whit these elements. There is no way to express fry-the-eggs in a single self contained expression thay you can reuse. As a consequence, OOP can not produce composable programs.
+## "Deconstructing the specification" recipe considered harmful ##
 
-In any case, in OOP, the programs are made of disconnected pieces, the software does not follow the natural flow as it would have been naturally extracted from the specifications and there are low level details. There will appear functionally uninteresting elements at the top level of the design, so the resulting code is hard to maintain.
+Usually the specification of some task can naturally be expressed as a process of step-by-step imperatives (often using verbs in second-person, active tense). People connect the relevant elements directly, without concern for other secondary elements. For example, a recipe would say, "You fry the eggs." It would not say, "There is fire and there are eggs; you start the fire and the eggs are fried by the fire." You see that a complete description in terms of active and passive elements implies low level elements that you are not interesting when writing a specification. But in OOP development, these elements must appear in the form of objects. Creating an OOP solution requires the deconstruction of the specification recipe into multiple "third-person-passive" perspectives (e.g. "it was fried" instead of "fry it"), one for each class or object that alternatively act as passive and active elements.
```

Comment:
I have a pretty hard time understanding this paragraph. It uses a lot of high-level concepts and I got lost without any concrete examples.